File: octo/msgp/errorlog-topology/src/main/java/com/sankuai/logparser/util/MtConfig.java
package com.sankuai.logparser.util;

import com.sankuai.meituan.config.MtConfigClient;
import com.sankuai.meituan.config.exception.MtConfigException;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

/**
 * Created by yves on 16/12/28.
 * Config utility for mt-config
 */
public class MtConfig {

    private static final Logger LOGGER = LoggerFactory.getLogger(MtConfig.class);
    private volatile MtConfigClient topologyCfgClient;
    private volatile MtConfigClient errorLogCfgClient;

    public MtConfig(){
        initTopologyCfg();
        initErrorLogCfg();
    }

    private static class MtConfigSingletonInstance {
        private static MtConfig INSTANCE = new MtConfig();
    }

    public static MtConfigClient getTopologyCfgClient() {
        return MtConfigSingletonInstance.INSTANCE.topologyCfgClient;
    }

    public static MtConfigClient getErrorLogCfgClient() {
        return MtConfigSingletonInstance.INSTANCE.errorLogCfgClient;
    }

    private void initTopologyCfg() {
        topologyCfgClient = new MtConfigClient();
        topologyCfgClient.setAppkey("com.sankuai.inf.octo.errorlog.topology");
        topologyCfgClient.setModel("v2");
        topologyCfgClient.setId("topologyCfg");
        LOGGER.info("topologyCfg client initialized, thread: " + Thread.currentThread().getName());
        try {
            topologyCfgClient.init();
        } catch (MtConfigException e) {
            LOGGER.error("init topologyCfgClient (" + topologyCfgClient.getNodeName()+ ") failed", e);
        }
    }

    private void initErrorLogCfg() {
        errorLogCfgClient = new MtConfigClient();
        errorLogCfgClient.setAppkey("com.sankuai.inf.octo.errorlog");
        errorLogCfgClient.setModel("v2");
        errorLogCfgClient.setId("errorLogCfg");
        LOGGER.info("errorLogCfg client initialized, thread: " + Thread.currentThread().getName());
        try {
            errorLogCfgClient.init();
        } catch (MtConfigException e) {
            LOGGER.error("init errorLogCfgClient (" + errorLogCfgClient.getNodeName()+ ") failed", e);
        }
    }
}

File: octo/oswatchRefactor/src/main/scala/com/sankuai/octo/oswatch/utils/MTConfig.scala
package com.sankuai.octo.oswatch.utils

import com.sankuai.meituan.config.MtConfigClient
import org.slf4j.{LoggerFactory, Logger}

/**
 * Created by dreamblossom on 15/9/29.
 */
object MTConfig {
  private final val LOG: Logger = LoggerFactory.getLogger(MTConfig.getClass)

  private val client = new MtConfigClient()
  private val instance = {
    client.setAppkey("com.sankuai.inf.octo.oswatch")
    //1.0.0及后面版本使用
    client.setModel("v2")
    client.setEnv("prod")
    client.setId("defaultClient")
    //可选，扫描注解的根目录，默认全部扫描
    client.setScanBasePackage("com.sankuai.meituan")

    try {
      // 初始化client
      client.init()
    } catch {
      case e: Exception => LOG.error("init MtConfigClient (" + client.getNodeName + ") failed", e);
    }
    client
  }

  def get(key: String, default: String = null) = {
    try {
      val value = instance.getValue(key)
      if (value != null) value else default
    } catch {
      case e: Exception => LOG.error(s"client get ${key} failed", e)
        default
    }
  }

  //  def main(args: Array[String]): Unit = {
  //    println(get("admins","default"))
  //  }
}
